byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -52,4 +70,22 @@ private String formatLocation(PlaceType location) {
     place = place.replace('_', ' ');
     return place;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String tripKeyString = request.getParameter("tripId");
+    Key tripKey = KeyFactory.stringToKey(tripKeyString);
+    ArrayList<String> interests = gson.fromJson(request.getReader(), ARRAYLIST_STRING);
+    Entity trip;
+    try {
+      trip = datastore.get(tripKey);
+    } catch(EntityNotFoundException e) {
+      //thrown automatically
+      logger.atInfo().withCause(e).log("Unable to find Trip Entity with key string %s", tripKeyString);
+      response.setStatus(response.SC_NOT_FOUND);
+      return;
+    }
+    trip.setProperty("interests", interests);
```

Comment:
It would be good to avoid working with the string property names outside the `Trip` class -- the point of having a wrapper class like that is to encapsulate the knowledge of what these string property names are, so that using them is less error prone. If you typo `setProperty("interessts", interests)` you'll wonder why in some places you can set the interests fine, and in others it doesn't work. If you typo `trip.setInteressts(interests)`, you'll get a compiler error, which is much easier to diagnose. In general string property access should be kept to as few places as possible, and should be encapsulated within the wrapper types. Have a look at rena's PR for some suggestions I gave on how to use the wrapper lasses for data manipulation like this (and coordinate among yourselves on who will implement them).

Reply:
Updated with the Trip class